In [3]:
import pandas as pd

In [4]:
#by_geo_district = pd.read_excel('./datasets/2020q3_where_workers_work.xls', sheet_name='Table8')

In [5]:
job2020_1 = pd.read_excel('./datasets/2020q3_where_workers_work.xls', sheet_name='Table9',
                         header=5,
                         nrows=1,
                         usecols='E,G,I,K')
job2020_2 = pd.read_excel('./datasets/2020q3_where_workers_work.xls', sheet_name='Table10',
                         header=5,
                         nrows=1,
                         usecols='E,G,I,K,M,O')
job2020_3 = pd.read_excel('./datasets/2020q3_where_workers_work.xls', sheet_name='Table11',
                         header=5,
                         nrows=1,
                         usecols='C,D,E,F,G,H,I,J,K,L,M')
job2020_4 = pd.read_excel('./datasets/2020q3_where_workers_work.xls', sheet_name='Table12',
                         header=5,
                         nrows=1,
                         usecols='C,D,E,F,G,H,I,J')
job2020_5 = pd.read_excel('./datasets/2020q3_where_workers_work.xls', sheet_name='Table13',
                         header=5,
                         nrows=1,
                         usecols='E,G,I,K,M')
job2020_6 = pd.read_excel('./datasets/2020q3_where_workers_work.xls', sheet_name='Table14',
                         header=5,
                         nrows=1,
                         usecols='C,D,E,F,G,H,I,J,K')
job2020_7 = pd.read_excel('./datasets/2020q3_where_workers_work.xls', sheet_name='Table15',
                         header=5,
                         nrows=1,
                         usecols='C,D,E,F,G,H,I,J,K')
job2020_8 = pd.read_excel('./datasets/2020q3_where_workers_work.xls', sheet_name='Table16',
                         header=5,
                         nrows=1,
                         usecols='D,F,H,J,L')

In [13]:
dataframes = [job2020_2, job2020_3, job2020_4, job2020_5, job2020_6, job2020_7, job2020_8]
jobs_2020 = job2020_1.join(dataframes)
jobs_2020.reset_index(inplace=True)
jobs_2020.rename(columns = {'index': 'zip_code', 0:'job_count'}, inplace=True)
jobs_2020['zip_code'].replace({"O'HARE":60666}, inplace=True)

In [19]:
job_data_zips = [jobs_2020['zip_code'][i] for i in range(len(jobs_2020))]

In [15]:
import folium
import json
from folium import plugins

In [16]:
with open('./datasets/zip_codes.geojson') as f:
    zips_map = json.load(f)

In [17]:
geojson_zips = sorted([int(zips_map['features'][i]['properties']['zip']) for i in range(len(zips_map['features']))])

In [18]:
set(geojson_zips) ^ set(job_data)

{60610, '60610 & 54', 60627, 60635, 60642, 60654, 60707, 60827}

- 60610: Cabrini-Green, Gold Coast (has a 1 block enclave in 60654? haha turns out it's the post office)
- 60654: River North
- 60627: Riverdale (neighborhood, not village. zip note showing up on google maps) - it has 60628
- 60827: Riverdale village (not part of Chicago)
- 60635: Elmwood Park (lmao they changed to 60707 in '96 https://www.chicagotribune.com/news/ct-xpm-1996-07-02-9607020235-story.html)
- 60707: Elmwood Park (not part of Chicago)
- 60642: Goose Island, Noble Square
- 60666: O'Hare (60656 is the weird split of O'Hare on the Chicago side)

In [11]:
len(zips['features'])

61

In [12]:
'''
getting rid of Riverdale (town) (60827) and Elmwood Park (60707)
from zipcode map since they're not part of Chicago
'''

for i in range(58):
    if zips['features'][i]['properties']['zip'] == '60707':
        del zips['features'][i]
    elif zips['features'][i]['properties']['zip'] == '60827':
        del zips['features'][i]

In [13]:
len(zips['features'])

58

In [14]:
len(by_zips)

57

In [19]:
'''
Dropping Elmwood Park (60635) from the job count dataframe because it's not part of Chicago.
Dropping Riverdale (neighborhood, aka zip code 60627) because this is no longer
a valid zip code. Also because there was nearly no job data on this neighborhood.
Based on reading through the data description for jobs, it seems that this data was 
split between 60628 (next to former 60627) and the town of Riverdale (60827).
'''

by_zips.drop(by_zips.loc[by_zips['zip_code'] == 60635].index, inplace = True)
by_zips.drop(by_zips.loc[by_zips['zip_code'] == 60627].index, inplace = True)
by_zips.reset_index(drop=True, inplace=True)

In [20]:
len(by_zips)

55

In [ ]:
len(by_zips)

## i'll put here which ones i'm gonna get rid of

In [ ]:
#zips['features'][i]['properties']['zip']

In [ ]:
type(zips['features'][0])

In [ ]:
chi_map = folium.Map(location=[41.881832, -87.623177], tiles='openstreetmap', zoom_start=11)

folium.GeoJson(zips).add_to(chi_map);

In [ ]:
chi_map

In [ ]:
geojson_zips = sorted([int(zips['features'][i]['properties']['zip']) for i in range(len(zips['features']))])
geojson_zips

In [ ]:
zips.keys()

In [ ]:
len(zips['features'])

In [ ]:
zips['features'][0].keys()

In [ ]:
zips['features'][0]['properties']['zip']

In [ ]:
geojson_zips = [zips['features'][i]['properties']['zip'] for i in range(len(zips['features']))]

In [ ]:
by_zip_1 = pd.read_excel('./datasets/2020q3_where_workers_work.xls', sheet_name='Table9',
                         #header=[3,4,5],
                         header=5,
                         nrows=1,
                         usecols='E,G,I,K')
by_zip_1

In [ ]:
with open('./datasets/zip_codes.geojson') as f:
    zips = json.load(f)

In [ ]:
chi_map = folium.Map(location=[41.881832, -87.623177], tiles='openstreetmap', zoom_start=11)

folium.Choropleth(
    geo_data=zips,
    name="choropleth",
    data=by_zip_1,
    #columns=[60603,60604,60601,60602],
    #key_on="features.properties",
    fill_color="YlGn",
    fill_opacity=0.7,
    line_opacity=0.2,
    legend_name="Number of Jobs",
).add_to(chi_map)

folium.LayerControl().add_to(chi_map);

In [ ]:
chi_map

In [ ]:
with open('./datasets/zip_codes.json') as f:
    zips = json.load(f)

In [ ]:
zips.keys()

In [ ]:
zips['data']

In [ ]:
chi_map = folium.Map(location=[41.881832, -87.623177], tiles='openstreetmap', zoom_start=11)

folium.Choropleth(
    geo_data=zips,
    #name="choropleth",
    #data=by_zip_1,
    #columns=[60603,60604,60601,60602],
    #key_on="features.properties",
    #fill_color="YlGn",
    #fill_opacity=0.7,
    #line_opacity=0.2,
    #legend_name="Number of Jobs",
).add_to(chi_map)

folium.LayerControl().add_to(chi_map);